zarr/core.py

```diff
@@ -443,6 +446,44 @@ def __getitem__(self, item):
         if not self._cache_metadata:
             self._load_metadata()
 
+        # handle zero-dimensional arrays
+        if self._shape == ():
+            return self._getitem_zd(item)
+        else:
+            return self._getitem_nd(item)
+
+    def _getitem_zd(self, item):
+        # special case __getitem__ for zero-dimensional array
+
+        # check item is valid
+        if item not in ((), Ellipsis):
+            raise IndexError('too many indices for array')
+
+        try:
+
+            # obtain encoded data for chunk
+            ckey = self._chunk_key((0,))
+            cdata = self.chunk_store[ckey]
+
+        except KeyError:
+
+            # chunk not initialized
+            out = np.empty((), dtype=self._dtype)
+            if self._fill_value is not None:
+                out.fill(self._fill_value)
+
+        else:
+
+            out = self._decode_chunk(cdata)
```

Comment:
Minor point. This seems a bit spaced out. Not sure if that was intentional or not.